ui/src/app/applications/components/application-node-info/application-node-info.tsx

```diff
@@ -167,8 +166,46 @@ export const ApplicationNodeInfo = (props: {
         });
     }
 
+    const getPodReadinessGatesState = React.useCallback(_getPodReadinessGatesState, [props.live]);
+
+    let ReadinessGatesFailedWarning: React.ReactNode = null;
+    if (props.live && props.node.kind === 'Pod') {
+        const readinessGatesState = getPodReadinessGatesState(props.live);
+
+        if (readinessGatesState.failedConditions.length > 0 || readinessGatesState.nonExistConditions.length > 0) {
+            ReadinessGatesFailedWarning = (
```

Comment:
IMO the code would read better if this were a separate component which received `readinessGatesState` as a prop, and was conditionally rendered inline based on the condition above. LMK if there's a specific reason why you went with this approach instead, either is okay 🙂

Reply:
Agreed. No specific reason, will move it to separate component